File: app/src/main/java/com/eddief/android/wallofaww/model/RedditPOJO.java
package com.eddief.android.wallofaww.model;

import com.google.gson.annotations.SerializedName;

public class RedditPOJO {

    @SerializedName("data")
    private RedditData data;

    public RedditData getData() {
        return data;
    }


    public class RedditData {

        @SerializedName("children")
        private RedditPost[] children;

        public RedditPost[] getChildren() {
            return children;
        }
    }

    public class RedditPost {

        @SerializedName("data")
        private RedditPostData data;


        public RedditPostData getData() {
            return data;
        }
    }
}

File: app/src/main/java/com/eddief/android/infiniteaww/model/RedditPOJO.kt
package com.eddief.android.infiniteaww.model

import com.google.gson.annotations.SerializedName

class RedditPOJO(
    @SerializedName("data")
    val data: RedditData
) {
    inner class RedditData(
        @SerializedName("children")
        val children: Array<RedditPost>
    ) {
        inner class RedditPost(
            @SerializedName("data")
            val data: RedditPostData
        )
    }
}